LabExT/Measurements/InsertionLossSweep.py

```diff
@@ -181,7 +207,63 @@ def algorithm(self, device, data, instruments, parameters):
         self.instr_laser.close()
         self.instr_pm.close()
 
+        # apply reference to data
+        if parameters['file path to reference meas.'].value.strip():
+            self.apply_reference_to_data(data=data)
+        else:
+            data['values']['referenced transmission [dB]'] = []
+
+        # if user wants to save disk space by discarding raw data, do it now
+        if parameters['discard raw transmission data'].value:
+            data['values']['transmission [dBm]'] = []
```

Comment:
@delmedico @boris-vukovic 

Check if you like the new implementation. I added another boolean parameter for the user to selectively discard the raw measurement data.

To note is that all save files now have the three data vectors (`wavelength [nm]`, `transmission [dBm]`, `referenced transmission [dB]`) always present as keys in the `data['values']` dict. If you did not chose to reference the data, the data is an empty list, similarly if you chose to discard the raw data.

Check if you like that solution and I'd be happy for some feedback.

Reply:
It's fine but I am not fully convinced by this approach. For one because now you always have to check the data arrays before reading some data and then based on that decide which one to further use (in post processing). And two, now in the main window plot you cannot simultaneously show calibrated and uncalibrated data. It would still be more natural to me if it is the calibrated data that is stored in `transmission [dB]` .

Reply:
So, we have discussed this in the last LabExT meeting. We unanimously agree that it is a terrible idea to have two traces with different meanings saved in the same vector name. You will have to adapt your analysis scripts.

And yes, the current implementation does not allow showing two differently named data vectors at the same time in the same plot axis. You can still open the extra plots window (double-click on measurement data plot) and view the two plots side-by-side.